src/ctelnet.cpp

```diff
@@ -1413,6 +1441,91 @@ void cTelnet::processTelnetCommand(const std::string& command)
     case TN_SB: {
         option = command[2];
 
+        // CHARSET
+        if (option == OPT_CHARSET && enableCHARSET) {
+            QByteArray payload = command.c_str();
+            if (payload.size() < 6) {
+                return;
+            }
+
+            // payload is in the Mud Server's encoding, trim off the Telnet suboption
+            // bytes from beginning (3) and end (2):
+            payload = payload.mid(3, static_cast<int>(payload.size()) - 5);
+
+            QString transcodedMsg;
+
+            if (mpOutOfBandDataIncomingCodec) {
+                // Message is encoded
+                transcodedMsg = mpOutOfBandDataIncomingCodec->toUnicode(payload);
+            } else {
+                // Message is in ASCII (though this can handle Utf-8):
+                transcodedMsg = QString::fromUtf8(payload);
+            }
+
+            // remove \r and \n from the data.
+            transcodedMsg.remove(QChar::CarriageReturn);
+            transcodedMsg.remove(QChar::LineFeed);
+            // replace ANSI escape character with escaped version, to handle improperly passed ANSI codes
+            transcodedMsg.replace(QLatin1String("\u001B"), QLatin1String("\\u001B"));
+
+            // CHARSET support per https://tools.ietf.org/html/rfc2066
+            if (command[3] == CHARSET_REQUEST) {
+                // No translate table support.  Discard. 
+                if (transcodedMsg.startsWith(QStringLiteral("[TTABLE]1"))) {
```

Comment:
Looking at **[RFC 2066](https://tools.ietf.org/html/rfc2066)** it seems that there is a space before the `]` in one example given but it is not present in a couple of others - so I am guessing it is a typo in that one case.

However I am not sure that it is permissible to just drop a `[TTABLE]` request here - as far as I can make out it must be explicitly rejected - but I cannot see how exactly... :frowning_face:

Reply:
I read the RFC many times and felt this was optional.  I agree that there is a typo there with that space.